src/dune_lang/action.ml

```diff
@@ -255,6 +358,19 @@ let decode_with_accepted_exit_codes =
             ]
 ;;
 
+let decode_rule =
+  let open Decoder in
+  let* targets = 
+    repeat (
+      let* target = String_with_vars.decode in
+      let* name = optional string in
```

Comment:
A tip with `let*` syntax is that whenever you have a `return` immediately after a `let*` you can convert the `let*` into `let+` and delete the `return`.